README.md

```diff
@@ -745,7 +746,26 @@ upload.status # => 'Your activity is ready.'
 upload.activity_id # => 1_998_557_443
 ```
 
-See [Strava::Models::Upload](lib/strava/models/upload.rb) for all available properties.
+```ruby
+# with error after upload
+begin
+  upload = client.upload(2136460097) 
+rescue Strava::Errors::UploadFailedError => e
+  e.status # => 200
+  e.error_status # => 'There was an error processing your activity.'
+  e.message # => '17611540601.tcx duplicate of activity 8309312818'
+  
+  e.upload_failed.external_id # => nil
+  e.upload_failed.activity_id # => nil
+  e.upload_failed.status # => 'There was an error processing your activity.'
+  e.upload_failed.error # => '17611540601.tcx duplicate of activity 8309312818'
+  e.upload_failed.class # => Strava::Models::UploadFailed
+end
+```
+
+See [Strava::Models::Upload](lib/strava/models/upload.rb) for all available properties.  
+See [Strava::Errors::UploadFailedError](lib/strava/errors/upload_failed_error.rb) for all available properties.
+See [Strava::Models::UploadFailed](lib/strava/models/upload_failed.rb) for all available properties.  
```

Comment:
Looks like `UploadFailed` and `UploadStatus` are the same thing. We don't need two identical classes that represent the same thing, so rename `e.upload_failed` to `e.upload_status`. Then make `UploadStatus#failed?`.